pkg/utils/queue/delaying_queue_test.go

```diff
@@ -104,9 +104,9 @@ func TestAddTwo(t *testing.T) {
 
 // checkLength checks whether the delay queue has the expected number of elements.
 // The check action is supposed to be instantaneous.
-// However, in order to reserve time for the internal synchronization of the delay queue itself,
+// However, in order to reserve time for the internal synchronization of the (delay) queue itself,
 // we perform the check in a polling manner and make it subject to a very short timeout(0.1s), the reserved time.
-func checkLength[T comparable](q DelayingQueue[T], len int) error {
+func checkLength[T comparable](q Queue[T], len int) error {
```

Comment:
```suggestion
func checkLength(q length, len int) error {
```

Define a generic interface for Len()
``` golang
type length interface {
	Len() int
}
```